src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseEventEnablingService.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.Map;
+import javax.inject.Inject;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.domain.enablingcondition.EnablingConditionParser;
+
+@Service
+public class CaseEventEnablingService {
+
+    private final EnablingConditionParser enablingConditionParser;
+
+    @Inject
+    public CaseEventEnablingService(EnablingConditionParser enablingConditionParser) {
+        this.enablingConditionParser = enablingConditionParser;
+    }
+
+    public Boolean evaluate(String enablingCondition,
+                           Map<String, JsonNode> caseEventData) {
```

Comment:
something weird here. Show condition can have metadata fields in them, i.e. [STATE]=someState. 
I would have thought you needed to pass the whole case so that metadata can be evaluated as well. Instead we are passing only the case fields in the data field

Reply:
If this issue is confirmed, can we also add some integration tests for event enabling conditions with metadata in show conditions to avoid regressions

Reply:
I'm thinking, do we have the same issue with case post states? Do we support metadata fields in show conditions there? we should...

Reply:
We are not supporting metadata fields in case post states for now...

Reply:
> something weird here. Show condition can have metadata fields in them, i.e. [STATE]=someState.
> I would have thought you needed to pass the whole case so that metadata can be evaluated as well. Instead we are passing only the case fields in the data field

changed to include CaseDetails and now supporting meta data as well in this..